### ISSUE TITLE
Missing features and ambiguities

### ISSUE THREAD

**Issue by cpacker** [cpacker/CLIPMesh#1]
## Missing features

Things described in the paper that are missing in this reimplementation:

- [ ] Multiple initial shape candidates
  - "_The initial shape for generation is selected from a set of basic primitives: horizontal or vertical cuboid and a sphere. Initially all shapes are rendered once with the same texture from multiple views and the shape with the lowest average score across the views is selected as an initial shape. This helps speed up training and leads to better mesh results since picking the right initial primitive can minimize the total deformation required._"

## Unresolved ambiguities

Things not described in the paper:

- Whether to apply the Laplacian regularizer to the vertices on the subdivided mesh or base mesh (paper figure implies it's on the base mesh)
- Base mesh details
  - How many vertices should be in the base/control shapes? (the current starting sphere may have too many)
- Subdivision details
  - How many subdivisions to do
  - Whether to optimize the displacement map of the subdivided mesh (or just the base vertices)
  - Currently, we're just using the base sphere provided in nvdiffmodeling
- Batch size, learning rate(s)
- Extras from CLIPMesh-SMPLX not mentioned in paper
  - Gaussian blurring of materials
  - Rendering without textures at the beginning of training (by default, 50% of the batch up to 50% of the epochs drop textures)
- Extras added not in paper or CLIPMesh-SMPLX
  - Negative tokens (eg "face" and "text")
